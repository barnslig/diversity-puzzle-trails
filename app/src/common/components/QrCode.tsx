import { Box, CircularProgress } from "@mui/material";
import { BrowserQRCodeSvgWriter } from "@zxing/browser";
import * as React from "react";

type QrCodeProps = {
  /**
   * Text to encode
   */
  contents: string;

  /**
   * Width of the SVG
   */
  width?: number;

  /**
   * Height of the SVG
   */
  height?: number;
};

/**
 * A QR-Code generator
 */
const QrCode = ({ contents, width = 300, height = 300 }: QrCodeProps) => {
  const codeRef = React.useRef<HTMLDivElement>(null!);
  const writer = React.useRef<BrowserQRCodeSvgWriter>(null!);

  const [isReady, setIsReady] = React.useState(false);

  React.useEffect(() => {
    (async () => {
      if (!writer.current) {
        const { BrowserQRCodeSvgWriter } = await import("@zxing/browser");
        writer.current = new BrowserQRCodeSvgWriter();
      }

      // Clear the div
      while (codeRef.current.firstChild) {
        codeRef.current.removeChild(codeRef.current.firstChild);
      }

      // Render the QR code
      writer.current.writeToDom(codeRef.current, contents, width, height);

      setIsReady(true);
    })();
  }, [contents, width, height]);

  return (
    <Box
      sx={{
        display: "flex",
      }}
    >
      <Box ref={codeRef} sx={{ fontSize: 0, background: "#fff" }} />

      {!isReady && (
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
            width,
            height,
          }}
        >
          <CircularProgress />
        </Box>
      )}
    </Box>
  );
};

export default QrCode;
