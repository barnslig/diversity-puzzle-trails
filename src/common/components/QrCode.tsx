import { makeStyles } from "@material-ui/core";
import { BrowserQRCodeSvgWriter } from "@zxing/browser";
import * as React from "react";

const useStyle = makeStyles((theme) => ({
  wrapper: {
    display: "flex",
  },
  code: {
    fontSize: 0,
    background: "#fff",
  },
}));

type QrCodeProps = {
  /**
   * Text to encode
   */
  contents: string;

  /**
   * Width of the SVG
   */
  width?: number;

  /**
   * Height of the SVG
   */
  height?: number;
};

/**
 * A QR-Code generator
 */
const QrCode = ({ contents, width = 300, height = 300 }: QrCodeProps) => {
  const classes = useStyle();

  const codeRef = React.useRef<HTMLDivElement>(null!);
  const writer = React.useRef(new BrowserQRCodeSvgWriter());

  React.useEffect(() => {
    // Clear the div
    while (codeRef.current.firstChild) {
      codeRef.current.removeChild(codeRef.current.firstChild);
    }

    // Render the QR code
    writer.current.writeToDom(codeRef.current, contents, width, height);
  }, [contents, width, height]);

  return (
    <div className={classes.wrapper}>
      <div ref={codeRef} className={classes.code} />
    </div>
  );
};

export default QrCode;
