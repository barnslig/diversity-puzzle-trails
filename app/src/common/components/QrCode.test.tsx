import { screen, waitForElementToBeRemoved } from "@testing-library/react";
import * as React from "react";
import render from "../testing/render";

import QrCode from "./QrCode";

it("renders correctly", async () => {
  const { container } = render(<QrCode contents="foobar" />);
  expect(container.firstChild).toMatchSnapshot();

  await waitForElementToBeRemoved(() => screen.getByRole("progressbar"));
  expect(container.firstChild).toMatchSnapshot();
});
