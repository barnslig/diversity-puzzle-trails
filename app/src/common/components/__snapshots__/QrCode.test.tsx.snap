// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="makeStyles-wrapper-1"
>
  <div
    class="makeStyles-code-2"
  />
  <div
    class="makeStyles-loading-3"
    style="width: 300px; height: 300px;"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  class="makeStyles-wrapper-1"
>
  <div
    class="makeStyles-code-2"
  >
    <svg
      height="300"
      width="300"
    >
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="55"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="95"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="145"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="45"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="55"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="55"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="55"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="145"
        y="55"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="55"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="55"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="55"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="145"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="155"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="65"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="75"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="145"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="155"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="85"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="95"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="95"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="95"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="95"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="145"
        y="95"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="95"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="95"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="55"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="95"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="145"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="105"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="115"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="55"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="115"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="145"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="155"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="125"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="95"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="115"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="155"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="175"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="135"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="155"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="145"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="55"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="95"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="115"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="175"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="155"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="55"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="145"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="155"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="165"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="175"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="175"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="175"
        y="175"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="175"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="175"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="175"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="175"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="175"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="175"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="55"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="95"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="155"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="185"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="195"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="195"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="195"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="195"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="195"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="145"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="175"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="205"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="175"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="215"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="215"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="155"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="175"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="185"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="195"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="245"
        y="225"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="235"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="235"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="235"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="135"
        y="235"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="155"
        y="235"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="165"
        y="235"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="205"
        y="235"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="45"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="55"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="65"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="75"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="85"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="95"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="105"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="125"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="175"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="225"
        y="245"
      />
      <rect
        fill="#000000"
        height="10"
        width="10"
        x="235"
        y="245"
      />
    </svg>
  </div>
</div>
`;
